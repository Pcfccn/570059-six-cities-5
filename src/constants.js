const Path = {
  MAIN: `/`,
  SIGN_IN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer/:id`,
  getUrlById: (id) => (`/offer/${id}`)
};

const ratingStars = [5, 4, 3, 2, 1];
const ratingInputTitle = [`perfect`, `good`, `not bad`, `badly`, `terribly`];

const widthStartsCoefficient = 20;

const CityCoordinates = {
  Amsterdam: [52.38333, 4.9],
  Paris: [48.864716, 2.349014],
};

const OfferCardClassName = {
  nearPlace: `near-places__card`,
  citiesPlace: `cities__place-card`,
};

const MapClassName = {
  property: `property__map`,
  cities: `cities__map`,
};

const cities = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export {Path, widthStartsCoefficient, ratingStars, ratingInputTitle, CityCoordinates, OfferCardClassName, MapClassName, cities};
