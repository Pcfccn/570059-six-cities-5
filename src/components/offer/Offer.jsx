import {PropTypes} from "prop-types";
import React from "react";
import {getMockNearestOffers, getOffer, getWidthForRatingStar} from "../../utils/common";
import PostCommentForm from "../offer-post-comment-form/offer-post-comment-form";
import Inside from "../offer-inside/offer-inside";
import Photos from "../offer-photos/offer-photos";
import Reviews from "../offer-reviews/offer-reviews";
import offerPropTypes from "../types/offer";
import MapComponent from "../map/map";
import {MapClassName, OfferCardClassName} from "../../constants";
import OfferCard from "../offer-card/offer-card";
import Header from "../header/header";
import {connect} from "react-redux";
import {reviews} from "../../mock/offer";

const Offer = ({offers, offerProps, onSubmitForm}) => {
  const offer = getOffer(offers, offerProps);
  const nearestOffers = getMockNearestOffers(offers, offerProps);
  const {image, isPremium, isInBookmarks, name, rating, type, bedrooms, adults, inside, host, price, location} = offer;
  const isInBookmarksButtonActive = isInBookmarks
    ? `property__bookmark-button--active`
    : ``;
  const hostClassName = host.pro
    ? `property__avatar-wrapper--pro`
    : ``;
  return (
    <div className="page">
      <Header />

      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              <Photos images={image} />
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {isPremium
                ? <div className="property__mark">
                  <span>Premium</span>
                </div>
                : ``}

              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {name}
                </h1>
                <button className={`property__bookmark-button ${isInBookmarksButtonActive} button`} type="button">
                  <svg className="place-card__bookmark-icon" width="31" height="33">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: `${getWidthForRatingStar(rating)}`}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{rating}</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {type}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {bedrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                    Max {adults} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{price.value}</b>
                <span className="property__price-text">&nbsp;{price.period}</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  <Inside inside={inside} />
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div className={`property__avatar-wrapper ${hostClassName} user__avatar-wrapper`}>
                    <img className="property__avatar user__avatar" src={host.avatar} width="74" height="74" alt="Host avatar"/>
                  </div>
                  <span className="property__user-name">
                    {host.name}
                  </span>
                </div>
                <div className="property__description">
                  <p className="property__text">
                    {host.description}
                  </p>
                </div>
              </div>
              <section className="property__reviews reviews">
                <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
                <ul className="reviews__list">
                  <Reviews reviews={reviews} />
                </ul>
                <PostCommentForm onSubmitForm={onSubmitForm}/>
              </section>
            </div>
          </div>
          <MapComponent
            className={MapClassName.PROPERTY}
            cityLocation={offer.cityLocation}
            zoom={offer.cityZoom}
            pinLocations={nearestOffers.map((it) => (it.location))}
            chosedPinLocation={location}
          />
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            <div className="near-places__list places__list">
              {nearestOffers.map((nearestOffer) => (
                <OfferCard
                  key={nearestOffer.id}
                  className={OfferCardClassName.NEAR_PLACE}
                  offer={nearestOffer}
                  onMouseEnterCard={()=>{}}
                />
              ))}
            </div>
          </section>
        </div>
      </main>
    </div>
  );
};

Offer.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    })
  }),
  reviews: PropTypes.arrayOf(
      PropTypes.shape({
        avatar: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        rating: PropTypes.number.isRequired,
        text: PropTypes.string.isRequired,
      })),
  offers: PropTypes.arrayOf(offerPropTypes).isRequired,
  onSubmitForm: PropTypes.func.isRequired,
  offerProps: PropTypes.object.isRequired,
};

const mapStateToProps = ({DATA}) => ({
  offers: DATA.offers,
});

export {Offer};
export default connect(mapStateToProps)(Offer);
