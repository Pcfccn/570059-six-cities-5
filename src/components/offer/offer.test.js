import React from "react";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import browserHistory from "../../brouser-history";
import {AuthorizationStatus, CommentSendingStatus} from "../../constants";
import {Offer} from "./Offer";
import configureMockStore from 'redux-mock-store';
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

const getOffersMock = (count) => {
  const templateOffers = Array(count)
    .fill(``)
    .map((_, index) => {
      return {
        id: index + 1,
        city: `Amsterdam`,
        reviews: [index + 2134, index + 2135],
        image: [`/img/apartment-01.jpg`, `/img/apartment-02.jpg`, `/img/apartment-03.jpg`, `/img/room.jpg`, `/img/studio-01.jpg`, `/img/studio-photos.jpg`],
        isPremium: Boolean(Math.random()),
        isInBookmarks: Boolean(Math.random()),
        name: `Beautiful & luxurious studio`,
        type: `Apartament`,
        bedrooms: 3,
        adults: 4,
        rating: 3.5,
        price: {
          value: 1234,
          period: `night`,
        },
        inside: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`,
          `Dishwasher`, `whatewer you want`],
        host: {
          avatar: `/img/avatar-angelina.jpg`,
          name: `Anagelina`,
          pro: Boolean(Math.random()),
          description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          The building is green and from 18th century.`,
        },
        location: [52.3909553943508, 4.85309666406198],
        previewImage: `/img/avatar-angelina.jpg`,
        cityLocation: [52.369553943508, 4.85309666406198],
        cityZoom: 10,
      };
    });

  if (count === 1) {
    const [oneOffer] = templateOffers;
    return oneOffer;
  }

  return templateOffers;
};
const noop = () => {};

configure({adapter: new Adapter()});

let mockState;
beforeEach(() => {
  mockState = {
    USER: {
      authorizationStatus: AuthorizationStatus.AUTH,
      email: `sdf@mail.ri`
    },
    STATE: {
      userComment: {
        rating: `5`,
        text: `xyyyyyyyyy`,
      },
      commentSendingStatus: CommentSendingStatus.DONE,
    },
  };
});

describe(`offer`, () => {
  it(`Should render correctly`, () => {
    const div = global.document.createElement(`div`);
    global.document.body.appendChild(div);

    const mockStore = configureMockStore();
    const store = mockStore(mockState);
    const wrapper = mount(
        <Provider store={store}>
          <Router history={browserHistory}>
            <Offer
              loadOfferInfo={noop}
              offerProps={{match: {params: {id: 1}}}}
              authorizationStatus={AuthorizationStatus.AUTH}
              offers={getOffersMock(2)}
              comments={[]}
              nearbyOffers={[getOffersMock(1)]}
            />
          </Router>
        </Provider>,
        {attachTo: div});

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
