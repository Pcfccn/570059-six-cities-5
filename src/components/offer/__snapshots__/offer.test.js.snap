// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offer Should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Offer
      authorizationStatus="AUTH"
      comments={Array []}
      loadOfferInfo={[Function]}
      nearbyOffers={
        Array [
          Object {
            "adults": 4,
            "bedrooms": 3,
            "city": "Amsterdam",
            "cityLocation": Array [
              52.369553943508,
              4.85309666406198,
            ],
            "cityZoom": 10,
            "host": Object {
              "avatar": "/img/avatar-angelina.jpg",
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          The building is green and from 18th century.",
              "name": "Anagelina",
              "pro": true,
            },
            "id": 1,
            "image": Array [
              "/img/apartment-01.jpg",
              "/img/apartment-02.jpg",
              "/img/apartment-03.jpg",
              "/img/room.jpg",
              "/img/studio-01.jpg",
              "/img/studio-photos.jpg",
            ],
            "inside": Array [
              "Wi-Fi",
              "Washing machine",
              "Towels",
              "Heating",
              "Coffee machine",
              "Baby seat",
              "Kitchen",
              "Dishwasher",
              "whatewer you want",
            ],
            "isInBookmarks": true,
            "isPremium": true,
            "location": Array [
              52.3909553943508,
              4.85309666406198,
            ],
            "name": "Beautiful & luxurious studio",
            "previewImage": "/img/avatar-angelina.jpg",
            "price": Object {
              "period": "night",
              "value": 1234,
            },
            "rating": 3.5,
            "reviews": Array [
              2134,
              2135,
            ],
            "type": "Apartament",
          },
        ]
      }
      offerProps={
        Object {
          "match": Object {
            "params": Object {
              "id": 1,
            },
          },
        }
      }
      offers={
        Array [
          Object {
            "adults": 4,
            "bedrooms": 3,
            "city": "Amsterdam",
            "cityLocation": Array [
              52.369553943508,
              4.85309666406198,
            ],
            "cityZoom": 10,
            "host": Object {
              "avatar": "/img/avatar-angelina.jpg",
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          The building is green and from 18th century.",
              "name": "Anagelina",
              "pro": true,
            },
            "id": 1,
            "image": Array [
              "/img/apartment-01.jpg",
              "/img/apartment-02.jpg",
              "/img/apartment-03.jpg",
              "/img/room.jpg",
              "/img/studio-01.jpg",
              "/img/studio-photos.jpg",
            ],
            "inside": Array [
              "Wi-Fi",
              "Washing machine",
              "Towels",
              "Heating",
              "Coffee machine",
              "Baby seat",
              "Kitchen",
              "Dishwasher",
              "whatewer you want",
            ],
            "isInBookmarks": true,
            "isPremium": true,
            "location": Array [
              52.3909553943508,
              4.85309666406198,
            ],
            "name": "Beautiful & luxurious studio",
            "previewImage": "/img/avatar-angelina.jpg",
            "price": Object {
              "period": "night",
              "value": 1234,
            },
            "rating": 3.5,
            "reviews": Array [
              2134,
              2135,
            ],
            "type": "Apartament",
          },
          Object {
            "adults": 4,
            "bedrooms": 3,
            "city": "Amsterdam",
            "cityLocation": Array [
              52.369553943508,
              4.85309666406198,
            ],
            "cityZoom": 10,
            "host": Object {
              "avatar": "/img/avatar-angelina.jpg",
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          The building is green and from 18th century.",
              "name": "Anagelina",
              "pro": true,
            },
            "id": 2,
            "image": Array [
              "/img/apartment-01.jpg",
              "/img/apartment-02.jpg",
              "/img/apartment-03.jpg",
              "/img/room.jpg",
              "/img/studio-01.jpg",
              "/img/studio-photos.jpg",
            ],
            "inside": Array [
              "Wi-Fi",
              "Washing machine",
              "Towels",
              "Heating",
              "Coffee machine",
              "Baby seat",
              "Kitchen",
              "Dishwasher",
              "whatewer you want",
            ],
            "isInBookmarks": true,
            "isPremium": true,
            "location": Array [
              52.3909553943508,
              4.85309666406198,
            ],
            "name": "Beautiful & luxurious studio",
            "previewImage": "/img/avatar-angelina.jpg",
            "price": Object {
              "period": "night",
              "value": 1234,
            },
            "rating": 3.5,
            "reviews": Array [
              2135,
              2136,
            ],
            "type": "Apartament",
          },
        ]
      }
    >
      <div
        className="page"
      >
        <Connect(Header)>
          <Header
            authStatus="AUTH"
            dispatch={[Function]}
            email="sdf@mail.ri"
          >
            <header
              className="header"
            >
              <div
                className="container"
              >
                <div
                  className="header__wrapper"
                >
                  <div
                    className="header__left"
                  >
                    <Link
                      className="header__logo-link"
                      to="/"
                    >
                      <LinkAnchor
                        className="header__logo-link"
                        href="/"
                        navigate={[Function]}
                      >
                        <a
                          className="header__logo-link"
                          href="/"
                          onClick={[Function]}
                        >
                          <img
                            alt="6 cities logo"
                            className="header__logo"
                            height="41"
                            src="/img/logo.svg"
                            width="81"
                          />
                        </a>
                      </LinkAnchor>
                    </Link>
                  </div>
                  <nav
                    className="header__nav"
                  >
                    <ul
                      className="header__nav-list"
                    >
                      <li
                        className="header__nav-item user"
                      >
                        <Link
                          className="header__nav-link header__nav-link--profile"
                          to="/favorites"
                        >
                          <LinkAnchor
                            className="header__nav-link header__nav-link--profile"
                            href="/favorites"
                            navigate={[Function]}
                          >
                            <a
                              className="header__nav-link header__nav-link--profile"
                              href="/favorites"
                              onClick={[Function]}
                            >
                              <div
                                className="header__avatar-wrapper user__avatar-wrapper"
                              />
                              <span
                                className="header__user-name user__name"
                              >
                                sdf@mail.ri
                              </span>
                            </a>
                          </LinkAnchor>
                        </Link>
                      </li>
                    </ul>
                  </nav>
                </div>
              </div>
            </header>
          </Header>
        </Connect(Header)>
        <main
          className="page__main page__main--property"
        >
          <section
            className="property"
          >
            <div
              className="property__gallery-container container"
            >
              <div
                className="property__gallery"
              >
                <Photos
                  images={
                    Array [
                      "/img/apartment-01.jpg",
                      "/img/apartment-02.jpg",
                      "/img/apartment-03.jpg",
                      "/img/room.jpg",
                      "/img/studio-01.jpg",
                      "/img/studio-photos.jpg",
                    ]
                  }
                >
                  <div
                    className="property__image-wrapper"
                    key="0-/img/apartment-01.jpg"
                  >
                    <img
                      alt="Photo studio"
                      className="property__image"
                      src="/img/apartment-01.jpg"
                    />
                  </div>
                  <div
                    className="property__image-wrapper"
                    key="1-/img/apartment-02.jpg"
                  >
                    <img
                      alt="Photo studio"
                      className="property__image"
                      src="/img/apartment-02.jpg"
                    />
                  </div>
                  <div
                    className="property__image-wrapper"
                    key="2-/img/apartment-03.jpg"
                  >
                    <img
                      alt="Photo studio"
                      className="property__image"
                      src="/img/apartment-03.jpg"
                    />
                  </div>
                  <div
                    className="property__image-wrapper"
                    key="3-/img/room.jpg"
                  >
                    <img
                      alt="Photo studio"
                      className="property__image"
                      src="/img/room.jpg"
                    />
                  </div>
                  <div
                    className="property__image-wrapper"
                    key="4-/img/studio-01.jpg"
                  >
                    <img
                      alt="Photo studio"
                      className="property__image"
                      src="/img/studio-01.jpg"
                    />
                  </div>
                  <div
                    className="property__image-wrapper"
                    key="5-/img/studio-photos.jpg"
                  >
                    <img
                      alt="Photo studio"
                      className="property__image"
                      src="/img/studio-photos.jpg"
                    />
                  </div>
                </Photos>
              </div>
            </div>
            <div
              className="property__container container"
            >
              <div
                className="property__wrapper"
              >
                <div
                  className="property__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="property__name-wrapper"
                >
                  <h1
                    className="property__name"
                  >
                    Beautiful & luxurious studio
                  </h1>
                  <Connect(BookmarksButton)
                    id={1}
                    isInBookmarks={true}
                    type="PROPERTY"
                  >
                    <BookmarksButton
                      addToBookmarks={[Function]}
                      id={1}
                      isInBookmarks={true}
                      type="PROPERTY"
                    >
                      <button
                        className="property__bookmark-button property__bookmark-button--active button"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          className="place-card__bookmark-icon"
                          height={33}
                          width={31}
                        >
                          <use
                            xlinkHref="#icon-bookmark"
                          />
                        </svg>
                        <span
                          className="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </BookmarksButton>
                  </Connect(BookmarksButton)>
                </div>
                <div
                  className="property__rating rating"
                >
                  <div
                    className="property__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "70%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                  <span
                    className="property__rating-value rating__value"
                  >
                    3.5
                  </span>
                </div>
                <ul
                  className="property__features"
                >
                  <li
                    className="property__feature property__feature--entire"
                  >
                    Apartament
                  </li>
                  <li
                    className="property__feature property__feature--bedrooms"
                  >
                    3
                     Bedrooms
                  </li>
                  <li
                    className="property__feature property__feature--adults"
                  >
                    Max 
                    4
                     adults
                  </li>
                </ul>
                <div
                  className="property__price"
                >
                  <b
                    className="property__price-value"
                  >
                    €
                    1234
                  </b>
                  <span
                    className="property__price-text"
                  >
                     
                    night
                  </span>
                </div>
                <div
                  className="property__inside"
                >
                  <h2
                    className="property__inside-title"
                  >
                    What's inside
                  </h2>
                  <ul
                    className="property__inside-list"
                  >
                    <Inside
                      inside={
                        Array [
                          "Wi-Fi",
                          "Washing machine",
                          "Towels",
                          "Heating",
                          "Coffee machine",
                          "Baby seat",
                          "Kitchen",
                          "Dishwasher",
                          "whatewer you want",
                        ]
                      }
                    >
                      <li
                        className="property__inside-item"
                        key="0-Wi-Fi"
                      >
                        Wi-Fi
                      </li>
                      <li
                        className="property__inside-item"
                        key="1-Washing machine"
                      >
                        Washing machine
                      </li>
                      <li
                        className="property__inside-item"
                        key="2-Towels"
                      >
                        Towels
                      </li>
                      <li
                        className="property__inside-item"
                        key="3-Heating"
                      >
                        Heating
                      </li>
                      <li
                        className="property__inside-item"
                        key="4-Coffee machine"
                      >
                        Coffee machine
                      </li>
                      <li
                        className="property__inside-item"
                        key="5-Baby seat"
                      >
                        Baby seat
                      </li>
                      <li
                        className="property__inside-item"
                        key="6-Kitchen"
                      >
                        Kitchen
                      </li>
                      <li
                        className="property__inside-item"
                        key="7-Dishwasher"
                      >
                        Dishwasher
                      </li>
                      <li
                        className="property__inside-item"
                        key="8-whatewer you want"
                      >
                        whatewer you want
                      </li>
                    </Inside>
                  </ul>
                </div>
                <div
                  className="property__host"
                >
                  <h2
                    className="property__host-title"
                  >
                    Meet the host
                  </h2>
                  <div
                    className="property__host-user user"
                  >
                    <div
                      className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
                    >
                      <img
                        alt="Host avatar"
                        className="property__avatar user__avatar"
                        height="74"
                        src="/img/avatar-angelina.jpg"
                        width="74"
                      />
                    </div>
                    <span
                      className="property__user-name"
                    >
                      Anagelina
                    </span>
                  </div>
                  <div
                    className="property__description"
                  >
                    <p
                      className="property__text"
                    >
                      A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          The building is green and from 18th century.
                    </p>
                  </div>
                </div>
                <section
                  className="property__reviews reviews"
                >
                  <h2
                    className="reviews__title"
                  >
                    Reviews · 
                    <span
                      className="reviews__amount"
                    />
                  </h2>
                  <ul
                    className="reviews__list"
                  >
                    <Reviews
                      reviews={Array []}
                    />
                  </ul>
                  <Connect(PostCommentForm)
                    id={1}
                  >
                    <PostCommentForm
                      changeCommentSendingStatus={[Function]}
                      enterText={[Function]}
                      id={1}
                      postComment={[Function]}
                      rating="5"
                      setRating={[Function]}
                      text="xyyyyyyyyy"
                    >
                      <form
                        action="#"
                        className="reviews__form form"
                        method="post"
                        onSubmit={[Function]}
                      >
                        <label
                          className="reviews__label form__label"
                          htmlFor="review"
                        >
                          Your review
                        </label>
                        <div
                          className="reviews__rating-form form__rating"
                        >
                          <RatingStar
                            currentRating="5"
                            key="5"
                            onChange={[Function]}
                            rating={5}
                          >
                            <input
                              checked={true}
                              className="form__rating-input visually-hidden"
                              id="5-stars"
                              name="rating"
                              onChange={[Function]}
                              type="radio"
                              value={5}
                            />
                            <label
                              className="reviews__rating-label form__rating-label"
                              htmlFor="5-stars"
                              title="perfect"
                            >
                              <svg
                                className="form__star-image"
                                height="33"
                                width="37"
                              >
                                <use
                                  xlinkHref="#icon-star"
                                />
                              </svg>
                            </label>
                          </RatingStar>
                          <RatingStar
                            currentRating="5"
                            key="4"
                            onChange={[Function]}
                            rating={4}
                          >
                            <input
                              checked={false}
                              className="form__rating-input visually-hidden"
                              id="4-stars"
                              name="rating"
                              onChange={[Function]}
                              type="radio"
                              value={4}
                            />
                            <label
                              className="reviews__rating-label form__rating-label"
                              htmlFor="4-stars"
                              title="good"
                            >
                              <svg
                                className="form__star-image"
                                height="33"
                                width="37"
                              >
                                <use
                                  xlinkHref="#icon-star"
                                />
                              </svg>
                            </label>
                          </RatingStar>
                          <RatingStar
                            currentRating="5"
                            key="3"
                            onChange={[Function]}
                            rating={3}
                          >
                            <input
                              checked={false}
                              className="form__rating-input visually-hidden"
                              id="3-stars"
                              name="rating"
                              onChange={[Function]}
                              type="radio"
                              value={3}
                            />
                            <label
                              className="reviews__rating-label form__rating-label"
                              htmlFor="3-stars"
                              title="not bad"
                            >
                              <svg
                                className="form__star-image"
                                height="33"
                                width="37"
                              >
                                <use
                                  xlinkHref="#icon-star"
                                />
                              </svg>
                            </label>
                          </RatingStar>
                          <RatingStar
                            currentRating="5"
                            key="2"
                            onChange={[Function]}
                            rating={2}
                          >
                            <input
                              checked={false}
                              className="form__rating-input visually-hidden"
                              id="2-stars"
                              name="rating"
                              onChange={[Function]}
                              type="radio"
                              value={2}
                            />
                            <label
                              className="reviews__rating-label form__rating-label"
                              htmlFor="2-stars"
                              title="badly"
                            >
                              <svg
                                className="form__star-image"
                                height="33"
                                width="37"
                              >
                                <use
                                  xlinkHref="#icon-star"
                                />
                              </svg>
                            </label>
                          </RatingStar>
                          <RatingStar
                            currentRating="5"
                            key="1"
                            onChange={[Function]}
                            rating={1}
                          >
                            <input
                              checked={false}
                              className="form__rating-input visually-hidden"
                              id="1-stars"
                              name="rating"
                              onChange={[Function]}
                              type="radio"
                              value={1}
                            />
                            <label
                              className="reviews__rating-label form__rating-label"
                              htmlFor="1-stars"
                              title="terribly"
                            >
                              <svg
                                className="form__star-image"
                                height="33"
                                width="37"
                              >
                                <use
                                  xlinkHref="#icon-star"
                                />
                              </svg>
                            </label>
                          </RatingStar>
                        </div>
                        <textarea
                          className="reviews__textarea form__textarea"
                          id="review"
                          maxLength={300}
                          name="review"
                          onChange={[Function]}
                          placeholder="Tell how was your stay, what you like and what can be improved"
                          required={true}
                          value="xyyyyyyyyy"
                        />
                        <div
                          className="reviews__button-wrapper"
                        >
                          <p
                            className="reviews__help"
                          >
                            To submit review please make sure to set 
                            <span
                              className="reviews__star"
                            >
                              rating
                            </span>
                             and describe your stay with at least 
                            <b
                              className="reviews__text-amount"
                            >
                              50 characters
                            </b>
                            .
                          </p>
                          <button
                            className="reviews__submit form__submit button"
                            disabled={true}
                            type="submit"
                          >
                            Submit
                          </button>
                        </div>
                      </form>
                    </PostCommentForm>
                  </Connect(PostCommentForm)>
                </section>
              </div>
            </div>
            <MapComponent
              chosedPinLocation={
                Array [
                  52.3909553943508,
                  4.85309666406198,
                ]
              }
              cityLocation={
                Array [
                  52.369553943508,
                  4.85309666406198,
                ]
              }
              className="property__map"
              key="1"
              pinLocations={
                Array [
                  Array [
                    52.3909553943508,
                    4.85309666406198,
                  ],
                ]
              }
              zoom={10}
            >
              <section
                className="property__map map"
                id="map"
              />
            </MapComponent>
          </section>
          <div
            className="container"
          >
            <section
              className="near-places places"
            >
              <h2
                className="near-places__title"
              >
                Other places in the neighbourhood
              </h2>
              <div
                className="near-places__list places__list"
              >
                <OfferCard
                  className="near-places__card"
                  key="1"
                  offer={
                    Object {
                      "adults": 4,
                      "bedrooms": 3,
                      "city": "Amsterdam",
                      "cityLocation": Array [
                        52.369553943508,
                        4.85309666406198,
                      ],
                      "cityZoom": 10,
                      "host": Object {
                        "avatar": "/img/avatar-angelina.jpg",
                        "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          The building is green and from 18th century.",
                        "name": "Anagelina",
                        "pro": true,
                      },
                      "id": 1,
                      "image": Array [
                        "/img/apartment-01.jpg",
                        "/img/apartment-02.jpg",
                        "/img/apartment-03.jpg",
                        "/img/room.jpg",
                        "/img/studio-01.jpg",
                        "/img/studio-photos.jpg",
                      ],
                      "inside": Array [
                        "Wi-Fi",
                        "Washing machine",
                        "Towels",
                        "Heating",
                        "Coffee machine",
                        "Baby seat",
                        "Kitchen",
                        "Dishwasher",
                        "whatewer you want",
                      ],
                      "isInBookmarks": true,
                      "isPremium": true,
                      "location": Array [
                        52.3909553943508,
                        4.85309666406198,
                      ],
                      "name": "Beautiful & luxurious studio",
                      "previewImage": "/img/avatar-angelina.jpg",
                      "price": Object {
                        "period": "night",
                        "value": 1234,
                      },
                      "rating": 3.5,
                      "reviews": Array [
                        2134,
                        2135,
                      ],
                      "type": "Apartament",
                    }
                  }
                  onMouseEnterCard={[Function]}
                >
                  <article
                    className="near-places__card place-card"
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <div
                      className="place-card__mark"
                    >
                      <span>
                        Premium
                      </span>
                    </div>
                    <div
                      className="cities__image-wrapper place-card__image-wrapper"
                    >
                      <Link
                        to="/offer/1"
                      >
                        <LinkAnchor
                          href="/offer/1"
                          navigate={[Function]}
                        >
                          <a
                            href="/offer/1"
                            onClick={[Function]}
                          >
                            <img
                              alt="Place image"
                              className="place-card__image"
                              height="200"
                              src="/img/avatar-angelina.jpg"
                              width="260"
                            />
                          </a>
                        </LinkAnchor>
                      </Link>
                    </div>
                    <div
                      className="place-card__info"
                    >
                      <div
                        className="place-card__price-wrapper"
                      >
                        <div
                          className="place-card__price"
                        >
                          <b
                            className="place-card__price-value"
                          >
                            €
                            1234
                            .
                          </b>
                          <span
                            className="place-card__price-text"
                          >
                            / 
                            night
                          </span>
                        </div>
                        <Connect(BookmarksButton)
                          id={1}
                          isInBookmarks={true}
                          type="PLACE_CARD"
                        >
                          <BookmarksButton
                            addToBookmarks={[Function]}
                            id={1}
                            isInBookmarks={true}
                            type="PLACE_CARD"
                          >
                            <button
                              className="place-card__bookmark-button place-card__bookmark-button--active button"
                              onClick={[Function]}
                              type="button"
                            >
                              <svg
                                className="place-card__bookmark-icon"
                                height={19}
                                width={18}
                              >
                                <use
                                  xlinkHref="#icon-bookmark"
                                />
                              </svg>
                              <span
                                className="visually-hidden"
                              >
                                To bookmarks
                              </span>
                            </button>
                          </BookmarksButton>
                        </Connect(BookmarksButton)>
                      </div>
                      <div
                        className="place-card__rating rating"
                      >
                        <div
                          className="place-card__stars rating__stars"
                        >
                          <span
                            style={
                              Object {
                                "width": "70%",
                              }
                            }
                          />
                          <span
                            className="visually-hidden"
                          >
                            Rating
                          </span>
                        </div>
                      </div>
                      <h2
                        className="place-card__name"
                      >
                        <Link
                          to="/offer/1"
                        >
                          <LinkAnchor
                            href="/offer/1"
                            navigate={[Function]}
                          >
                            <a
                              href="/offer/1"
                              onClick={[Function]}
                            >
                              Beautiful & luxurious studio
                            </a>
                          </LinkAnchor>
                        </Link>
                      </h2>
                      <p
                        className="place-card__type"
                      >
                        Apartament
                      </p>
                    </div>
                  </article>
                </OfferCard>
              </div>
            </section>
          </div>
        </main>
      </div>
    </Offer>
  </Router>
</Provider>
`;
